docs/content/docs/dev/table/materialized-table/syntax.md

```diff
@@ -0,0 +1,337 @@
+CREATE MATERIALIZED TABLE my_materialized_table
+    PARTITIONED BY (ds)
+    WITH (
+        'format' = 'json',
+        'partition.fields.ds.date-formatter' = 'yyyy-MM-dd'
+    )
+    ...
+```
+As shown in the above example, we specified the date-formatter parameter for the ds partition field. During actual scheduling, the scheduling time will be converted to the ds partition value. For example, for a scheduling time of 2024-01-01 00:00:00, only the partition with ds='2024-01-01' will be updated.
+
+<span class="label label-danger">Note</span>
+- The field in the [time format parameters]({{< ref "docs/dev/table/config" >}}#partition-fields-date-formatter) must be a valid partition field and must be of string type.
+
+## Examples
+
+Create a materialized table with a data freshness of 10 seconds and a refresh mode of CONTINUOUS:
+
+```sql
+CREATE MATERIALIZED TABLE my_materialized_table_continuous
+    PARTITIONED BY (ds)
+    WITH (
+        'format' = 'json',
+        'partition.fields.ds.date-formatter' = 'yyyy-MM-dd'
+    )
+    FRESHNESS = INTERVAL '10' SECOND
+    REFRESH_MODE = 'CONTINUOUS'
+    AS 
+    SELECT 
+        k.ds,
+        k.user_id,
+        COUNT(*) AS event_count,
+        SUM(k.amount) AS total_amount,
+        MAX(u.age) AS max_age
+    FROM 
+        kafka_catalog.db1.kafka_table k
+    JOIN 
+        user_catalog.db1.user_table u
+    ON 
+        k.user_id = u.user_id
+    WHERE 
+        k.event_type = 'purchase'
+    GROUP BY 
+        k.ds, k.user_id
+```
+
+Create a materialized table with a data freshness of 10 minutes and a refresh mode of FULL:
+
+```sql
+CREATE MATERIALIZED TABLE my_materialized_table_full
+    PARTITIONED BY (ds)
+    WITH (
+        'format' = 'json',
+        'partition.fields.ds.date-formatter' = 'yyyy-MM-dd'
+    )
+    FRESHNESS = INTERVAL '10' MINUTE
+    REFRESH_MODE = 'FULL'
+    AS 
+    SELECT 
+        p.ds,
+        p.product_id,
+        p.product_name,
+        AVG(s.sale_price) AS avg_sale_price,
+        SUM(s.quantity) AS total_quantity
+    FROM 
+        paimon_catalog.db1.product_table p
+    LEFT JOIN 
+        paimon_catalog.db1.sales_table s
+    ON 
+        p.product_id = s.product_id
+    WHERE 
+        p.category = 'electronics'
+    GROUP BY 
+        p.ds, p.product_id, p.product_name
+```
+
+## Limitations
+
+- Does not support explicitly specifying columns
+- Does not support modified query statements
+- Does not support using temporary tables, temporary views, or temporary functions in the select query
+
+# ALTER MATERIALIZED TABLE
+
+```
+ALTER MATERIALIZED TABLE [catalog_name.][db_name.]table_name SUSPEND | RESUME [WITH (key1=val1, key2=val2, ...)] | REFRESH [PARTITION partition_spec]
+```
+
+ALTER MATERIALIZED TABLE is used to manage materialized tables. This command allows users to suspend and resume refresh tasks of materialized tables and manually trigger data refreshes.
```

Comment:
ALTER MATERIALIZED TABLE is used to manage materialized tables. This command allows users to suspend and resume refresh pipeline of materialized tables and manually trigger data refreshes.

## SUSPEND

```
ALTER MATERIALIZED TABLE [catalog_name.][db_name.]table_name SUSPEND
```

SUSPEND is used to pause the background refresh pipeline of the materialized table.

**Example:**

```sql
-- Specify SAVEPOINT path before pausing
SET 'execution.checkpointing.savepoint-dir' = 'hdfs://savepoint_path';

-- Suspend the specified materialized table
ALTER MATERIALIZED TABLE my_materialized_table SUSPEND;
```

<span class="label label-danger">Note</span>
- When suspending a table in CONTINUOUS mode, the job will be paused using STOP WITH SAVEPOINT by default. You need to set the SAVEPOINT save path using [parameters]({{< ref "docs/deployment/config" >}}#execution-checkpointing-savepoint-dir).

## RESUME

```
ALTER MATERIALIZED TABLE [catalog_name.][db_name.]table_name RESUME [WITH (key1=val1, key2=val2, ...)]
```

RESUME is used to resume the refresh pipeline of the materialized table.

**Example:**

```sql
-- Resume the specified materialized table
ALTER MATERIALIZED TABLE my_materialized_table RESUME;

-- Resume the specified materialized table and specify sink parallelism
ALTER MATERIALIZED TABLE my_materialized_table RESUME WITH ('sink.parallelism'='10');
```

## REFRESH

```
ALTER MATERIALIZED TABLE [catalog_name.][db_name.]table_name REFRESH [PARTITION partition_spec]
```

REFRESH is used to proactively trigger the refresh of the materialized table.

**Example:**

```sql
-- Refresh the entire table data
ALTER MATERIALIZED TABLE my_materialized_table REFRESH;

-- Refresh specified partition data
ALTER MATERIALIZED TABLE my_materialized_table REFRESH PARTITION (ds='2024-06-28');
```

<span class="label label-danger">Note</span>
- The REFRESH operation will start a Flink batch job to refresh the materialized table data.

# DROP MATERIALIZED TABLE

```text
DROP MATERIALIZED TABLE [IF EXISTS] [catalog_name.][database_name.]table_name
```

When deleting a materialized table, the background refresh pipeline will be stopped first, and then the metadata corresponding to the materialized table will be removed from the Catalog.

**Example:**

```sql
-- Delete the specified materialized table
DROP MATERIALIZED TABLE IF EXISTS my_materialized_table;
```